File: backend/src/main/java/com/andrew/hdss/dtos/CreateQuestionRequest.java
package com.andrew.hdss.dtos;// package: match your existing dto package

import com.andrew.hdss.models.enums.MappedEntity;
import com.andrew.hdss.models.enums.QuestionType;

public record CreateQuestionRequest(
        String name,
        String label,
        String hint,
        QuestionType type,
        boolean required,
        String relevant,
        String constraint,
        String constraintMessage,
        String calculation,
        String choiceListName,
        MappedEntity mappedEntity,
        String mappedField
) {}

File: frontend/src/models/CreateQuestionRequest.ts
import type { MappedEntity } from "./types/MappedEntity";
import type { QuestionType } from "./types/QuestionTypes";

export interface CreateQuestionRequest{
    name: string;
    label: string;
    hint: string;
    type: QuestionType;
    required: boolean;
    relevant: string;
    constraint: string;
    constraintMessage: string;
    calculation: string;
    choiceListName: string;
    mappedEntity: MappedEntity;
    mappedField: string;
}